rust/osm2lanes-bin/src/main.rs

```diff
@@ -0,0 +1,62 @@
+use std::fs::File;
+use std::io::BufReader;
+use std::path::PathBuf;
+
+use clap::{AppSettings, Parser, Subcommand};
+use futures::executor::block_on;
+use osm2lanes::overpass::get_way;
+use osm2lanes::tags::Tags;
+use osm2lanes::{tags_to_lanes, Locale, TagsToLanesConfig};
+
+/// Simple program to greet a person
+#[derive(Parser, Debug)]
+#[clap(name = "osm2lanes", author, version, about, long_about = None)]
+struct Cli {
+    #[clap(short, long, parse(from_occurrences))]
+    verbose: usize,
+    #[clap(subcommand)]
+    command: Command,
+}
+
+#[derive(Subcommand, Debug)]
+enum Command {
+    /// Retrieve lanes given OSM way ID
+    #[clap(setting(AppSettings::ArgRequiredElseHelp))]
+    Way {
+        /// Way ID
+        #[clap(required = true)]
+        id: u64,
+    },
+    /// Convert OSM way tags to lanes
+    #[clap(setting(AppSettings::ArgRequiredElseHelp))]
+    Convert {
+        /// JSON of OSM Tags
+        #[clap(required = true, parse(from_os_str))]
+        path: PathBuf,
+    },
+}
+
+#[tokio::main]
+async fn main() {
```

Comment:
I tried to look for a lightweight non-spinning async executor that only supports `block_on`, but I had no luck....

Reply:
It sort of feels like tokio is emerging as the standard now. I think with the right feature flags, it shouldn't add too much compile time or final binary size